src/commands/scan.ts

```diff
@@ -0,0 +1,139 @@
+import {Command, flags} from '@oclif/command';
+import {Scanner, ScanResult} from '../scanner';
+import {ResourceGraphTarget, DummyTarget, Target, RuleType} from '../rules';
+import cli from 'cli-ux';
+import chalk = require('chalk');
+
+interface LogOptions {
+  color?: string;
+  indent?: number;
+  bold?: boolean;
+}
+
+export default class Scan extends Command {
+  private isDebugMode = false;
+  private isVerbose = false;
+
+  static description =
+    'Scans Azure resources for potential configuration issues';
+
+  static examples = [
+    `$ aza scan --scope <SCOPE>
+    [rule-name]
+        [✓ | ❌][rule-description]     
+        Resources:
+                [resource-ids]
+
+    [total-passing]
+    [total-failing]
+    [total-rules-scanned]   
+`,
+  ];
+
+  static flags = {
+    help: flags.help({char: 'h'}),
+    scope: flags.string({
+      char: 's',
+      description: 'Azure subscription id to scan',
+    }),
+    dummy: flags.boolean({
```

Comment:
no-op (don't spend time on this): in cases like this, I think we should model this as just another type, ex: `aza scan --type dummy` (and then we'll delete it later) versus having a dedicated "dummy" flag